### PR TITLE
Add bison.

### PR DESCRIPTION
**Pull request by isaachier**

### CODE REVIEW

File: cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+    GLOBAL_INSTALL_DIR
+      "@HUNTER_INSTALL_PREFIX@"
+    PARALLEL_JOBS
+      "@HUNTER_JOBS_OPTION@"
+    PACKAGE_CONFIGURATION_TYPES
+      "@HUNTER_PACKAGE_CONFIGURATION_TYPES@"
+    EXTRA_FLAGS
+      --disable-rpath
```
Comment (ruslo) [inline cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in]: No need to introduce new scheme just for `EXTRA_FLAGS`. You can add `--disable-rpath` as CMAKE_ARGS, example:
* https://github.com/ruslo/hunter/blob/7799c576559549714362fd59900676a6bf4564b2/cmake/projects/PostgreSQL/hunter.cmake#L36-L40

File: docs/packages/pkg/bison.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.. spelling::
+
+    bison
+
+.. index:: compiler ; bison
+
+.. _pkg.foo:
```
Comment (ruslo) [inline docs/packages/pkg/bison.rst]: -> `_pkg.bison`

File: cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in (hunk 1 above)
Comment (isaachier) [reply cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in]: Oh from the documentation I thought any autogen package required required at least one custom scheme.

File: docs/packages/pkg/bison.rst (hunk 2 above)
Comment (isaachier) [reply docs/packages/pkg/bison.rst]: Good catch thanks.

File: cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in (hunk 1 above)
Comment (ruslo) [reply cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in]: > Oh from the documentation I thought any autogen package required required at least one custom scheme

No, it's not. Feel free to send the patch with update to clarify this.

Comment (isaachier) [reply cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in]: So instead I should copy paste the custom scheme into this file? The content there seems necessary to invoke `./configure` etc.

Comment (isaachier) [reply cmake/projects/bison/schemes/url_sha1_bison_autotools.cmake.in]: Ignore my question. Will use PostgreSQL as guide here.

**Comment by ruslo** [comment]
Some toolchains failed. To exclude broken please send pull request to [pkg.bison](https://github.com/ingenue/hunter/tree/pkg.bison).

**Comment by ruslo** [comment]
Please rebase/squash to resolve conflicts.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2426
* https://travis-ci.org/ingenue/hunter/builds/276973941?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
> Testing

Some toolchains failed.

Excluding instructions:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#excluding-tests

**Comment by isaachier** [comment]
Actually this is an interesting problem in iOS build:

```
Performing install step for 'bison-universal'

cd /Users/travis/build/ingenue/hunter/_testing/Hunter/_Base/48efcab/76d000c/df2ea98/Build/bison/Source/universal && /Users/travis/build/ingenue/hunter/_ci/cmake/bin/cmake -P /Users/travis/build/ingenue/hunter/_testing/Hunter/_Base/48efcab/76d000c/df2ea98/Build/bison/Source/universal/autotools-merge-lipo.cmake

CMake Error at autotools-merge-lipo.cmake:234 (message):

  lipo execution failed: fatal error: lipo: can't figure out the architecture

  type of:

  /Users/travis/build/ingenue/hunter/_testing/Hunter/_Base/48efcab/76d000c/df2ea98/Build/bison/Build/multi-arch/armv7/bin/yacc
```

If you look at `bin/yacc` in any bison installation, you will see it is a shell script to reroute to bison binary. I wonder what the specific issue is here.

**Comment by ruslo** [comment]
> Actually this is an interesting problem in iOS build

iOS universal build is not supported well for autotools.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2435
* https://travis-ci.org/ingenue/hunter/builds/277475809?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
No upload since package is not cacheable. Released: https://github.com/ruslo/hunter/releases/tag/v0.19.106

**Pull request merged**